src/release.rs

```diff
@@ -306,3 +396,113 @@ pub fn produce_install_only(tar_zst_path: &Path) -> Result<PathBuf> {
 
     Ok(dest_path)
 }
+
+pub fn produce_install_only_stripped(tar_gz_path: &Path, llvm_dir: &Path) -> Result<PathBuf> {
+    let buf = std::fs::read(tar_gz_path)?;
+
+    let size_before = buf.len();
+
+    let gz_data = convert_to_stripped(
+        std::io::Cursor::new(buf),
+        std::io::Cursor::new(vec![]),
+        llvm_dir,
+    )?
+    .into_inner();
+
+    let size_after = gz_data.len();
+
+    println!(
+        "stripped {} from {size_before} to {size_after} bytes",
+        tar_gz_path.display()
+    );
+
+    let filename = tar_gz_path
+        .file_name()
+        .expect("should have filename")
+        .to_string_lossy();
+
+    let mut name_parts = filename
+        .split('-')
+        .map(|x| x.to_string())
+        .collect::<Vec<_>>();
+    let parts_len = name_parts.len();
+
+    name_parts[parts_len - 1] = "install_only_stripped".to_string();
+
+    let install_only_name = name_parts.join("-");
+    let install_only_name = format!("{install_only_name}.tar.gz");
+
+    let dest_path = tar_gz_path.with_file_name(install_only_name);
+    std::fs::write(&dest_path, gz_data)?;
+
+    Ok(dest_path)
+}
+
+/// URL from which to download LLVM.
+///
+/// To be kept in sync with `pythonbuild/downloads.py`.
+static LLVM_URL: Lazy<Url> = Lazy::new(|| {
+    if cfg!(target_os = "macos") {
+        if std::env::consts::ARCH == "aarch64" {
+            Url::parse("https://github.com/indygreg/toolchain-tools/releases/download/toolchain-bootstrap%2F20240222/llvm-17.0.6+20240222-aarch64-apple-darwin.tar.zst").unwrap()
+        } else if std::env::consts::ARCH == "x86_64" {
+            Url::parse("https://github.com/indygreg/toolchain-tools/releases/download/toolchain-bootstrap%2F20240222/llvm-17.0.6+20240222-x86_64-apple-darwin.tar.zst").unwrap()
+        } else {
+            panic!("unsupported macOS architecture");
+        }
+    } else if cfg!(target_os = "linux") {
+        Url::parse("https://github.com/indygreg/toolchain-tools/releases/download/toolchain-bootstrap%2F20240222/llvm-17.0.6+20240222-gnu_only-x86_64-unknown-linux-gnu.tar.zst").unwrap()
+    } else {
+        panic!("unsupported platform");
+    }
+});
+
+/// Bootstrap `llvm` for the current platform.
+///
+/// Returns the path to the top-level `llvm` directory.
+pub async fn bootstrap_llvm() -> Result<PathBuf> {
+    let url = &*LLVM_URL;
+    let filename = url.path_segments().unwrap().last().unwrap();
+
+    let llvm_dir = PathBuf::from("llvm");
+
+    // If `llvm` is already available with the target version, return it.
```

Comment:
We can also just store this in a temporary directory for the duration of the command. It's a little tedious when iterating locally, since LLVM is large and you then have to re-download every time, but I don't feel strongly about keeping this.